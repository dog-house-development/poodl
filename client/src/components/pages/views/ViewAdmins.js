import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import { fetchAdmins } from '../../../actions/adminActions';
import DataGrid from '../../ui/DataGrid';

export class ViewAdmins extends Component {
    componentDidMount() {
        // call redux action to retrieve all admins from api
        this.props.getAdmins();
    }

    getDataGridContent() {
        // choose what we want to display out of the admins data
        let data = [];
        _.each(this.props.admins, admin => {
            // we want admins' names and emails, and we need a key which will not be displayed
            data.push({ name: admin.name, email: admin.email, key: admin._id });
        });
        return data;
    }

    render() {
        return (
            <div className="view-all-container">
                <h1>View All Admins</h1>
                <DataGrid data={this.getDataGridContent()} loading={this.props.loading} />
            </div>
        );
    }
}

export const mapStateToProps = (state, props) => {
    return {
        admins: state.admins.all.data,
        loading: state.admins.loading,
        errors: state.errors
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        getAdmins: () => dispatch(fetchAdmins())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ViewAdmins);
