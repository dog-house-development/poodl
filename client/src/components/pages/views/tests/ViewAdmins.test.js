import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import _ from 'lodash';

import { ViewAdmins, mapStateToProps, mapDispatchToProps } from '../ViewAdmins';

configure({ adapter: new Adapter() });

describe('ViewAdmins tests', () => {
    let wrapper, instance;
    const setInstanceAndWrapper = (_props = {}, _state = {}) => {
        const state = _.assign(
            {},
            {
                admins: {
                    loading: false,
                    all: {
                        data: [
                            {
                                _id: '123',
                                name: 'Big Tup',
                                email: 'bigtup@nowhere.com'
                            },
                            {
                                _id: '321',
                                name: 'Lil Tup',
                                email: 'liltup@nowhere.com'
                            }
                        ]
                    }
                },
                errors: {}
            },
            _state
        );
        const props = _.assign({}, _props);
        wrapper = shallow(
            <ViewAdmins
                {..._.assign(
                    {},
                    props,
                    mapStateToProps(state, props),
                    mapDispatchToProps(jasmine.createSpy('dispatch'))
                )}
            />
        );
        instance = wrapper.instance();
    };

    beforeEach(() => {
        setInstanceAndWrapper();
    });

    describe('getDataGridContent', () => {
        it('should return filtered admins data', () => {
            spyOn(instance, 'getDataGridContent');
            instance.getDataGridContent();
            expect(instance.getDataGridContent()).toBe([
                { name: 'Big Tup', email: 'bigtup@nowhere.com', key: 123 },
                { name: 'Lil Tup', email: 'liltup@nowhere.com', key: 321 }
            ]);
        });
    });

    describe('render', () => {
        it('should render correctly', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
