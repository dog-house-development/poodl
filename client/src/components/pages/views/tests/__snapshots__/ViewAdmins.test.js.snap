// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAdmins tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewAdmins
    admins={
      Array [
        Object {
          "_id": "123",
          "email": "bigtup@nowhere.com",
          "name": "Big Tup",
        },
        Object {
          "_id": "321",
          "email": "liltup@nowhere.com",
          "name": "Lil Tup",
        },
      ]
    }
    errors={Object {}}
    getAdmins={[Function]}
    loading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          View All Admins
        </h1>,
        <DataGrid
          data={
            Array [
              Object {
                "email": "bigtup@nowhere.com",
                "key": "123",
                "name": "Big Tup",
              },
              Object {
                "email": "liltup@nowhere.com",
                "key": "321",
                "name": "Lil Tup",
              },
            ]
          }
          loading={false}
        />,
      ],
      "className": "view-all-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View All Admins",
        },
        "ref": null,
        "rendered": "View All Admins",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "email": "bigtup@nowhere.com",
              "key": "123",
              "name": "Big Tup",
            },
            Object {
              "email": "liltup@nowhere.com",
              "key": "321",
              "name": "Lil Tup",
            },
          ],
          "loading": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            View All Admins
          </h1>,
          <DataGrid
            data={
              Array [
                Object {
                  "email": "bigtup@nowhere.com",
                  "key": "123",
                  "name": "Big Tup",
                },
                Object {
                  "email": "liltup@nowhere.com",
                  "key": "321",
                  "name": "Lil Tup",
                },
              ]
            }
            loading={false}
          />,
        ],
        "className": "view-all-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View All Admins",
          },
          "ref": null,
          "rendered": "View All Admins",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "email": "bigtup@nowhere.com",
                "key": "123",
                "name": "Big Tup",
              },
              Object {
                "email": "liltup@nowhere.com",
                "key": "321",
                "name": "Lil Tup",
              },
            ],
            "loading": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
